components/uniset/src/uniset.rs

```diff
@@ -0,0 +1,328 @@
+use std::{boxed::Box, error::Error, iter::Iterator, slice::Iter, vec::Vec};
+
+const CODEPOINT_MAX: u32 = 0x10FFFF; // does max imply inclusive? else should be 10FFFF
+const CODEPOINT_MIN: u32 = 0x000000;
+const BMP_MAX: u32 = 0xFFFF;
+
+/// Given string representation of inversion list create set
+///
+/// See UnicodeSet::new for conditions
+fn parse_serial_string(serialize_str: &str) -> Result<Vec<u32>, Box<dyn Error>> {
```

Comment:
I am not sure if this is possible here since the types of errors that can be encountered are varied. Please correct me if there is a way to return std::error::Error there besides for defining our own Error types with `impl Error`. I think even if we define our own Error types the benefits are not great if they are limited to this function.

Reply:
AFAIK there are ways to unify error handling around `std::error::Error` provided that all relevant error types either support it or can be `Into`'d.  I also *think* that it would be useful to unify error handling as the zoo of errors you can get while working on Unicode is considerable and good diagnostic messages can be really helpful for debugging.

For example, https://github.com/dtolnay/thiserror and https://github.com/dtolnay/anyhow allow fairly painless hook-up of new errors. Though I don't know if it's `no_std` friendly.

I'd personally be in favor of defining own local error types.  Thoughts?

Reply:
We have not yet decided on an error library to use. I wouldn't mind coalescing around `thiserror` (which is intended for libraries, `anyhow` is intended for client apps), but I'm ok delaying that decision a bit and using `dyn Error` for now.

Reply:
Ah, interesting.  I created #153 to capture the point.